securedrop_client/queue.py

```diff
@@ -28,6 +28,36 @@
 logger = logging.getLogger(__name__)
 
 
+class RunnablePriorityQueue(PriorityQueue):
+    """
+    Wrapper class around PriorityQueue that emits a signal when message or reply
+    download jobs are enqueued or dequeued.
+    """
+
+    def __init__(self, *args: Any, queue_updated_signal: pyqtSignal = None, **kwargs: Any):
```

Comment:
```suggestion
    def __init__(self, *args: Any, queue_updated_signal: Optional[pyqtSignal] = None, **kwargs: Any):
```
I am surprised that **mypy** didn't catch the implicit `Optional`. :shrug: 


> Explicit is better than implicit. -- [PEP 20][PEP20] - The Zen of Python

  [PEP20]: https://peps.python.org/pep-0020/

Reply:
After reading the section in the docs on [None and Optional handling](https://mypy.readthedocs.io/en/stable/command_line.html#none-and-optional-handling) this surprised me too! It sounds like the default behavior changed as of version 0.980 so that parameters with `None` as a default value are no longer treated as Optional be default. 

I think what's going on is that mypy can't find a library stub for `PyQt5.QtCore` so in the absence of a type for `pyqtSignal`, it's being treated as `Any`. Since `None` is a subtype of `Any`, mypy doesn't raise an "incompatible default for argument" error.
It looks like there is a [stub library for PyQt5](https://pypi.org/project/PyQt5-stubs/) which it might be worth looking into adding to the project! (I'm happy to raise an issue or PR on this) 
In the meantime I'll use the Optional type here in the interest of making the intended type more clear.

Reply:
Yes, we recently upgraded **mypy** and got that change (https://github.com/freedomofpress/securedrop-client/pull/1599).

I think your hypothesis is very very likely to be correct. It makes a lot of sense to me at least, and I'll give a try to the stub library. Thanks for doing the research @zekehuntergreen! :star2:

Reply:
That's it! You're correct.

Adding the **pyqt5-stub** library surfaces a series of type errors and warnings. (Expected.)

I'll open a separate PR for it, as it will likely take a little bit of time to fix them all. But I can confirm that **mypy** notices implicitly optional `pyqtSignal`s now :rocket:

Reply:
By the way, the signals we're passing as arguments might actually be of type `pyqtBoundSignal`! :bulb:

Reply:
Amazing, I'll take a look at #1611